tests/functional/services/compute/v3/test_register_endpoint_v3.py

```diff
@@ -0,0 +1,39 @@
+import uuid
+
+import globus_sdk
+from globus_sdk._testing import load_response
+
+ENDPOINT_CONFIG = """
+display_name: My Endpoint
+engine:
+  type: GlobusComputeEngine
+"""
+
+
+def test_register_endpoint(compute_client_v3: globus_sdk.ComputeClientV3):
+    load_response(compute_client_v3.register_endpoint)
+    request_doc = {
+        "endpoint_name": "my_endpoint",
+        "display_name": "My Endpoint",
+        "version": "2.31.0",
+        "multi_user": False,
+        "allowed_functions": [str(uuid.uuid1())],
+        "authentication_policy": str(uuid.uuid1()),
+        "subscription_uuid": str(uuid.uuid1()),
+        "metadata": {
+            "endpoint_config": ENDPOINT_CONFIG.strip(),
+            "user_config_template": "",
+            "user_config_schema": {},
```

Comment:
Given that this is testing the happy path, perhaps we don't include the fields not relevant to this non-MEP registration request?  Certainly these two fields are not necessary (right?  I haven't double-checked just now, maybe I'm missing something).  `multi_user` is similarly not required here.

In terms of the need for this test's purpose, I'm not sure about some of the other fields either.  (Have to run at this time; let me know if I'm missing the thought-process.)

Reply:
I'm fine trimming the example payload down, but I also see benefit to showing all possible fields.

Reply:
As mentioned outside of this thread, I don't see the value-add.  Who's the audience for this test data and slightly-bloated interaction?  To my understanding, this is test data ... for tests.

It's just a test, so I'm not overly chuffed about it, but I do observe that it doesn't match what we expect the real-world interaction to be, even if it does work.  If it's to flesh out edge cases, then we have more tests to write to do just that.  But if it's just to exercise the `globus_sdk` code and new function, then I'd gently encourage a more expected data set.

Regardless, this observation is not a blocker.

Reply:
From previous conversations, I understood these to partially serve as examples for usage (e.g., [test_collections.py](https://github.com/globus/globus-sdk-python/blob/8df9c5cac8845180ca03085f03bd985da6087b65/tests/functional/services/gcs/test_collections.py#L88-L113)). If that's wrong, then I agree with limiting the request payload to what is necessary for the test.

@kurtmckee, can you provide some clarification?

Reply:
I'll bet that it's been said that these test documents are examples for usage, but I wouldn't target the tests for that purpose; I would focus on excellent documentation for that purpose.